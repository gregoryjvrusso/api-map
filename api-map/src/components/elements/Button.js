import styled, { css } from 'styled-components'
import { ifProp } from 'styled-tools'

const warning = css`
  background-color: ${p => p.theme.colors.backgrounds.warning.none};
  border-color: ${p => p.theme.colors.backgrounds.warning.none}

  :hover{
    border-color: ${p => p.theme.colors.backgrounds.warning.hover}
  }
`
const primary = css`
  background-color: ${p => p.theme.colors.backgrounds.primary.blue};
  border-color: ${p => p.theme.colors.backgrounds.primary.blue}

  :hover{
    border-color: ${p => p.theme.colors.backgrounds.primary.hover}
  }
`

const Button = styled.button`
  font-size: 14px;
  margin: 4px;
  padding: 8px 14px;
  border: 2px solid;
  border-radius: 3px;
  font-weight: bold;
  color: #FFFFFF;
  ${ifProp('warning', warning)}
  ${ifProp('primary', primary)}
`

export default Button
