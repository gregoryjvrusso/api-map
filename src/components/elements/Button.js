import styled, { css } from 'styled-components'
import { ifProp } from 'styled-tools'
import { width } from 'styled-system'

const primary = css`
  background-color: ${p => p.theme.colors.backgrounds.primary.blue};
  border: 0px;
`

const Button = styled.button`
  font-size: 14px;
  margin: 4px;
  padding: 8px 14px;
  border: 2px solid;
  border-radius: 3px;
  font-weight: bold;
  color: #FFFFFF;
  ${ifProp('primary', primary)}
  ${width}
`

export default Button
