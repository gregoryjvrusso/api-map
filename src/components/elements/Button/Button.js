import styled from 'styled-components'
import { width } from 'styled-system'
import { theme } from 'styled-tools'

const Button = styled.button`
  font-size: 14px;
  margin: 4px;
  padding: 8px 14px;
  border: 2px solid;
  border-radius: 3px;
  font-weight: bold;
  color: ${theme('colors.texts.white')};
  background-color: ${theme('colors.backgrounds.primary.blue')};
  ${width}
`

export default Button
