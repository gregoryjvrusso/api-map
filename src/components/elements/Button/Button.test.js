import React from 'react'
import renderer from 'react-test-renderer'
import Button from './Button.js'

describe('Button component', () => {
  it('deve conseguir renderizar [snapshot]', async () => {
    const render = renderer
    .create(<Button />)
    .toJSON()
    expect(render).toMatchSnapshot()
  })
})
