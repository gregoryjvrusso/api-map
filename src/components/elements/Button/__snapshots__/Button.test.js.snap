// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component deve conseguir renderizar [snapshot] 1`] = `
<button
  className="sc-bdVaJa fPEQVc"
/>
`;
